### DISCUSSION TITLE
Conversation Retrival QA Chain- return source documents- Showing JSON only

### DISCUSSION THREAD

**Discussion by rickystanley76** [FlowiseAI/Flowise#3546] [Q&A]
![image](https://github.com/user-attachments/assets/583c01e0-4385-4bcc-9854-ddf78e6977d4)
I have this flow for my RAG system using Cohere, it's working pretty well. Only trouble i am having is the Return Source Document is only showing JSON(as below), where i would like to show as the part of the document. For Embedding is used cohere multilingual and few PDF documents. I would like to show the parts of the PDF as source documents. 
![image](https://github.com/user-attachments/assets/a70b440d-49cf-4f17-855e-99ce7d803f40)
Can anyone suggest some tips/solutions? 
Thanks in advance.

**Comment by hcancelik** [comment]
@rickystanley76 where you able to figure it out? I'm also trying to parse some information from the metadata but couldn't able to get it.

**Reply by rickystanley76** [reply]
Hello @hcancelik no not yet. I have also written on their discord but still no responses.  :(

**Comment by Malak-Alaabiad** [comment]
@rickystanley76 I'm also trying to parse some metadata from the document source but couldn't able to get it.

**Reply by rickystanley76** [reply]
@Malak-Alaabiad still i am waiting for the solutions. let's if there are someone from flowise can see this and help.
